chore: Add browser name and version in mixpanel payload

prajwalnl opened the pull request:
## Type of Change

<!-- Put an `x` in the boxes that apply -->

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

<!-- Describe your changes in detail -->

- Added browser name and version in mixpanel payload.
- Fixed merchantName field to use actual merchant name from merchant details instead of user's name.
- Fixed name field to use the user's name instead of email.

## Motivation and Context

<!--
Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.

If you don't have an issue, we'd recommend starting with one first so the PR
can focus on the implementation (unless its an obvious bug or documentation fix
that will have little conversation).
-->

## How did you test it?

<!--
Did you write an integration/unit/API test to verify the code changes?
Or did you test this change manually (provide relevant screenshots)?
-->

## Where to test it?

- [ ] INTEG
- [ ] SANDBOX
- [ ] PROD

## Checklist

<!-- Put an `x` in the boxes that apply -->

- [ ] I ran `npm run re:build`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

prajwalnl committed:
chore: add browser name and version in mixpanel payload

prajwalnl committed:
chore: debug hyperswitch pg

prajwalnl committed:
Merge branch 'main' into mixpanel-browser-name

prajwalnl committed:
chore: use getValueFromArray

prajwalnl committed:
chore: update browserTypes

prajwalnl committed:
Merge branch 'main' into mixpanel-browser-name

prajwalnl committed:
chore: address comments

prajwalnl committed:
chore: Address comments

prajwalnl committed:
fix: fix payload key values

prajwalnl committed:
chore: remove docker log

Muditbhatia12 commented on src/libraries/Window.res:
```diff
@@ -195,6 +204,81 @@ module Navigator = {
+      if parts->Array.length > 1 {
+        let rest = parts->LogicUtils.getValueFromArray(1, "")
+        let versionParts = rest->String.split(" ")
+        if versionParts->Array.length > 0 {
```
LogicUtils can be opened once at the top

prajwalnl commented on src/hooks/MixpanelHook.res:
```diff
@@ -61,6 +61,7 @@ let useSendEvent = () => {
         "lang": Navigator.browserLanguage,
         "$os": Navigator.platform,
         "$browser": Navigator.browserName,
```
Made code changes

prajwalnl committed:
refactor: simplify Window.Navigator browser detection using polymorphic variants

- Convert browserType from regular to polymorphic variants enabling :> string coercion
- Introduce browserEntries data-driven lookup table unifying UA matching for both
  browser name detection and version extraction
- Remove redundant browserNameRaw external binding (appName was unused)
- Simplify browserNameToString: special-case only #InternetExplorer, all others
  use polymorphic coercion
- open LogicUtils at Navigator module level for getValueFromArray access

Note: IE11 (Trident-only UA) version extraction now returns '' instead of
  the rv: version — IE11 is not a supported browser target.
